common/test/test_lint.py

```diff
@@ -1,22 +1,104 @@
+# SPDX-FileCopyrightText: © 2023 Christian BUHTZ <c.buhtz@posteo.jp>
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In Time" which is released under GNU
+# General Public License v2 (GPLv2).
+# See file LICENSE or go to <https://www.gnu.org/licenses/#GPL>.
+"""Tests using several linters.
+
+Instead of failing tests are skipped on machines where the linters are not
+available. As an exception only on TravisCI tests are forced to run and will
+fail if linters not available.
+"""
 import unittest
 import os
 import pathlib
 import subprocess
 import shutil
 from typing import Iterable
+try:
+    import pycodestyle
+    PYCODESTYLE_AVAILABLE = True
+except ImportError:
+    PYCODESTYLE_AVAILABLE = False
+
+BASE_REASON = ('Using package {0} is mandatory on TravisCI, on '
+               'other systems it runs only if `{0}` is available.')
 
 ON_TRAVIS = os.environ.get('TRAVIS', '') == 'true'
-PYLINT_AVIALBE = not shutil.which('pylint') is None
-PYLINT_REASON = ('Using PyLint is mandatory on TravisCI, on other systems'
-                 'it runs only if `pylint` is available.')
+TRAVIS_REASON = ('Running linter tests is mandatory on TravisCI only. On '
+                 'other machines they will run only if linters available.')
+
+PYLINT_AVAILABLE = shutil.which('pylint') is not None
+RUFF_AVAILABLE = shutil.which('ruff') is not None
+
+ANY_LINTER_AVAILABLE = any((
+    PYLINT_AVAILABLE,
+    RUFF_AVAILABLE,
+    PYCODESTYLE_AVAILABLE,
+))
 
+# Files in this lists will get the full battery of linters and rule sets.
+full_test_files = [pathlib.Path(fp) for fp in (
+    'bitbase.py',
+    'schedule.py',
+    'version.py',
+    'test/test_lint.py',
+)]
 
+# Not all linters do respect PEP8
+PEP8_MAX_LINE_LENGTH = 79
+
+
+def create_pylint_cmd(include_error_codes=None):
+    """Create the pylint base command for later use with subprocess.run().
+
+    Args:
+        include_error_codes: List of exclusive rules to check for. If empty all
+            default rules are checked.
+    """
+
+    cmd = [
+        'pylint',
+        # Make sure BIT modules can be imported (to detect "no-member")
+        '--init-hook=import sys;'
+        'sys.path.insert(0, "./../qt");'
+        'sys.path.insert(0, "./../common");',
+        # Storing results in a pickle file is unnecessary
+        '--persistent=n',
+        # autodetec number of parallel jobs
+        '--jobs=0',
+        # Disable scoring  ("Your code has been rated at xx/10")
+        '--score=n',
+        # prevent false-positive no-module-member errors
+        '--extension-pkg-allow-list=PyQt6,PyQt6.QtCore',
+        # Because of globally installed GNU gettext functions
+        '--additional-builtins=_,ngettext',
+        # PEP8 conform line length (see PyLint Issue #3078)
+        '--max-line-length=79',
```

Comment:
If you keep the options here in the Python script, you'll want to use the variable defined above for `max-line-length`:
```suggestion
        f'--max-line-length={PEP8_MAX_LINE_LENGTH}',
```